test/bdd/steps/api/search-assertion.js

```diff
@@ -0,0 +1,76 @@
+const { When, Then, Given } = require('@cucumber/cucumber');
+const { expect, assert } = require('chai');
+const utilities = require('../../../utilities/utilities');
+let handlerId, searchResult, searchData, parsedKeywords, publishData;
+
+Given(/^I call search request on node (\d+) with ([^"]*) for the keywords:*$/, { timeout: 120000 }, async function (node, resultType, keywords) {
+    resultType = 'assertions';
+    parsedKeywords = utilities.unpackRawTableToArray(keywords);
+
+    const result = await this.state.nodes[node - 1].client
+        .search(resultType, parsedKeywords[0]).catch((error) => {
+            assert.fail(`Error while trying to search assertion. ${error}`);
+        });
+
+    expect(result.status).to.be.eql(202);
+    expect(result.statusText).to.be.eql('Accepted');
+    expect(result.data.handler_id).to.be.a('string');
+    handlerId = result.data.handler_id;
+
+    if (handlerId === undefined) {
+        assert.fail(`Error while trying to get search result for ${parsedKeywords[0]}, handler_id is ${handlerId}`);
+    }
+    this.state.lastPublishData = {
+        nodeId: node - 1,
+        handlerId,
+        keywords: parsedKeywords,
+        assertion: '',
+    };
+});
+When('I wait searching request to be finalized', { timeout: 120000 }, async function () {
+    this.logger.log('I wait searching request to be finalized');
+    publishData = this.state.lastPublishData;
+
+        this.logger.log(`Getting search result for handler id: ${publishData.handlerId} on node: ${publishData.nodeId}`);
+        // eslint-disable-next-line no-await-in-loop
+        searchResult = await this.state.nodes[publishData.nodeId].client.getResult(publishData.handlerId, 'assertions:search').catch((error) => {
```

Comment:
After fetching the search result you can save it in state:
this.state.searchData.result = searchResult
And than in next steps use the data from state